models/Asset.php

```diff
@@ -1659,6 +1659,7 @@ public function addMetadata($name, $type, $data = null, $language = null)
     {
         if ($name && $type) {
             $tmp = [];
+            $name = str_replace("~", '-', $name);
```

Comment:
just wondering if you try to replace a~b with a-b but a-b already exists

Reply:
changed replacement to "---" so now it will be an edge case!